torch/csrc/autograd/input_buffer.cpp

```diff
@@ -82,6 +82,23 @@ bool can_accumulate_inplace(const Variable& v) {
       at::caching::adjusted_use_count(v) == 1 && v.has_storage() &&
       v.storage().use_count() == 1);
 }
+
+void sync_streams(
+    std::optional<c10::Stream>& stream_be_synced,
+    std::optional<c10::Stream>& sync_stream,
+    Variable& var,
+    const c10::DeviceType device_type) {
+  TORCH_INTERNAL_ASSERT(sync_stream.has_value());
```

Comment:
If sync_stream always has a value in this function, its type should be c10::Stream&.

Reply:
> If sync_stream always has a value in this function, its type should be `c10::Stream&` or `const c10::Stream&`. The runtime check should be moved to compile time check and the caller has to dereference their optional stream.

Do you mean that we pass argument to the function by "stream.value()". but it looks like we also need a check to guarantee "stream.value()" is not null, or it will crash without any notification.
BTW, how does compile time check work here ?